R/lint.R

```diff
@@ -218,14 +218,14 @@ lint_dir <- function(path = ".", relative_path = TRUE, ..., exclusions = list("r
   lints <- flatten_lints(lapply(
     files,
     function(file) {
-      if (interactive()) {
+      if (interactive() && !identical(Sys.getenv("TESTTHAT"), "true")) {
```

Comment:
I couldn't get `mockery::stub()` to work here.
Trying `mockery::stub(lint_dir, "interactive", FALSE)` didn't work as intended.
Trying `mockery::stub(lint_package, "interactive", FALSE, depth = 2)` worked, but messed up the global state so much that following tests couldn't even find `rex::re_matches` etc.

In the end, I thought it's probably useful to silence `lint_dir()` during testthat runs automatically.

Reply:
Filed #972 as a follow-up. I think it's worth investing more thought into what and how to control output.
The "." output is quite daunting at times especially when linting a large codebase with many files.
`cli` could do a lot regarding output, or `progressr`.